src/App.js

```diff
@@ -4,42 +4,72 @@ import './App.css';
 import Hero from './ui/Hero'
 import Feature from './entities/Feature'
 import FeatureFilter from './ui/FeatureFilter'
+import CounterPickFilter from './ui/CounterPickFilter';
 import arrayToTable from './util/arrayToTable'
 import sortBy from './util/sortBy'
 import mapToObjects from './util/mapToObjects'
+const mergeObjects = (a, b) => Object.assign({}, a, b)
 
 class App extends Component {
   state = {
-    filterSelection: {}
+    filterSelection: {},
+    filterCounterPickSelection: {},
+    features: [],
+    heroes: [],
+    counterPicks: [],
+    moreThen2FilterMode: false
   }
 
   componentDidMount() {
     axios.get('/features.json')
          .then(response => this.setState({ features: sortBy(mapToObjects(response.data, Feature), 'title') }));
     axios.get('/heroes.json')
          .then(response => this.setState({ heroes: sortBy(response.data, 'name') }));
+    axios.get('/counter-pick.json')
+         .then(response => this.setState({ counterPicks: sortBy(response.data, 'title') }));
   }
 
-  render() {
-    const {filterSelection, features} = this.state;
-    const mergeObjects = (a, b) => Object.assign({}, a, b)
+  updateFilters = (c) => {
```

Comment:
Please, use ES6 method syntax like with `render()`

If you don't want create lambda functions for callbacks you can use `.bind` - it's pretty common practice AFAIK.

(I prefer slow lambdas though)